### DISCUSSION TITLE
WebGL2 and Tutorials

### DISCUSSION THREAD

**Discussion by Antix-Development** [KilledByAPixel/LittleJS#45] [Ideas]
An upgrade to WebGL2 for NPOT support would be good and also some tutorials so you didn't have to try deciphering somebody else's code would be fantastic.

**Comment by KilledByAPixel** [comment]
NOPT textures are already supported but we may upgrade to WebGL2 eventually. A the time it was written there was less overall support.

What kind of tutorial are you looking for, a text write up or video? There are many example projects that are easy to understand. Would a walkthrough of one of those examples help?

Unlike most engines, the code is intended to be looked at. It is very clean and short with many comments and documentation.

**Comment by Antix-Development** [comment]
Thanks for the speedy reply. I didn't know NPOT textures were already supported, I don't think I read that anywhere.

As for tutorials I was thinking of just simple ones like how to draw and move a sprite around, and maybe how to create a tilemap.

Having some small examples in the documentation would be great because the documentation its self seems kind of flat and sterile, but that's kind of the nature of generated documents I think.

However, I'll be less lazy and start reading the code since those tutorials won't just appear over-night if at all :)

Awesome engine BTW, and I always use ZzFX in my JS13K games :)

**Comment by KilledByAPixel** [comment]
That's a good idea. My plan for a while has been to make a short video tutorial to walk people through one of the example projects. I will try to get to that soon, definitely before next JS13K.

The documentation needs work for sure, it's kind of on the back burner at the moment.

Thanks! I will leave this one open to remind me to make a video tutorial.

**Comment by Antix-Development** [comment]
Awesome, thanks. Next JS13k is in August ;)

**Comment by chriscamacho** [comment]
+1 on this, just api docs and a few examples, make for a less than accessible process, looking for something to teach a 12 y/o coding don't know if this would be suitable due to lack of tutorials...

**Comment by KilledByAPixel** [comment]
Hey, thanks for the encouragement. I wrote up a full tutorial for how to make a simple game with LittleJS. This walks through the entire process from scratch and only takes about 30 minutes. Would be great to hear feedback from someone who tries it. All the code is included so it is suitable even for people who are very new to programming.

https://github.com/KilledByAPixel/LittleJS/tree/main/examples/breakoutTutorial

**Comment by chriscamacho** [comment]
Thats a very good start! Bare in mind all the api docs in the world won't tell someone how different functions are supposed to be used together, especially for novice coders, if you want more adoption of this library I would strongly recommend making more tutorials !

**Comment by KilledByAPixel** [comment]
Thanks, let me know if you give the tutorial a try. I really worked on this one to keep it easy to understand while showing a lot.

You are right about needing more tutorials, what kind do you think would help the most?

**Comment by Antix-Development** [comment]
Tried the tutorial, great stuff!

If you were going to make more I'd suggest subjects like creating tile maps, from Tiled data and also from user supplied 2d arrays.

**Comment by KilledByAPixel** [comment]
I implemented WebGL2 a while ago and created a tutorial so I am going to close this out. There is also an issue to add more tutorials here...

https://github.com/KilledByAPixel/LittleJS/issues/47
